Neos.ContentRepository/Classes/Domain/Model/Node.php

```diff
@@ -237,14 +237,15 @@ protected function isNodePathAvailable(string $path): bool
     protected function setPathInternal(string $destinationPath, bool $recursiveCall): array
     {
         $originalPath = $this->nodeData->getPath();
+        $childNodes = $this->getChildNodes();
+
+        $this->moveNodeToDestinationPath($this, $destinationPath);
```

Comment:
When an aggregate is moved there is a check that the children moved to an existing parent, so the parent and children are moved together.

This is the other way around for non-aggregates as the children are moved first (for unknown reasons to me) and then the parent which will make the integrity check fail when an aggregate is inside a non aggregate as the children are then disconnected until the parent is moved. IMO this could maybe explain some other issues we had in the past of disconnected children. See the problem described in https://github.com/neos/neos-development-collection/issues/3340#issuecomment-1673068873

Reply:
Hmm sorry i cant follow your argument @Sebobo ... could you please elaborate further?

Reply:
Discussed via Slack and agreed upon that we need to add some tests